lambdas/process-journey-event/src/main/java/uk/gov/di/ipv/core/processjourneyevent/statemachine/StateMachineInitializer.java

```diff
@@ -26,25 +29,35 @@ public class StateMachineInitializer {
             new ObjectMapper(new YAMLFactory()).configure(STRICT_DUPLICATE_DETECTION, true);
     private static final ObjectMapper om = new ObjectMapper();
     private Map<String, State> journeyStates;
-    private Map<String, NestedJourneyDefinition> nestedJourneyDefinitions;
+    private final HashMap<String, NestedJourneyDefinition> nestedJourneyDefinitions =
+            new HashMap<>();
     private final StateMachineInitializerMode mode;
+    private final List<String> nestedJourneyTypes;
 
     public StateMachineInitializer(IpvJourneyTypes journeyType) {
-        this(journeyType, StateMachineInitializerMode.STANDARD);
-    }
-
-    public StateMachineInitializer(IpvJourneyTypes journeyType, StateMachineInitializerMode mode) {
+        this(
+                journeyType,
+                StateMachineInitializerMode.STANDARD,
+                Stream.of(NestedJourneyTypes.values())
+                        .map(NestedJourneyTypes::getJourneyName)
+                        .toList());
+    }
+
+    public StateMachineInitializer(
+            IpvJourneyTypes journeyType,
+            StateMachineInitializerMode mode,
+            List<String> nestedJourneyTypes) {
         this.journeyType = journeyType;
         this.mode = mode;
+        this.nestedJourneyTypes = nestedJourneyTypes;
     }
 
     private final IpvJourneyTypes journeyType;
 
     public Map<String, State> initialize() throws IOException {
         Journey journey = yamlOm.readValue(getJourneyConfig(journeyType), new TypeReference<>() {});
         journeyStates = journey.states();
-        nestedJourneyDefinitions =
-                yamlOm.readValue(getNestedJourneyDefinitionsConfig(), new TypeReference<>() {});
+        getNestedJourneysFromConfig();
```

Comment:
Is there a reason you decided to switch this around, rather than `nestedJourneyDefinitions = getNestedJourneysFromConfig()`?

It doesn't really matter - just curious!

Reply:
but I've updated this now to make it more consistent with `journeyStates` :)